### PR TITLE
Ctest define enum

### PR DESCRIPTION
**Pull request by dcristoloveanu**
Ctest define enum

### CODE REVIEW

File: doc/ctest_user_manual.md [hunk 1]
```diff
@@ -305,3 +305,35 @@ The `CTest` harness supports out of the box string formatting and comparers for
 - `double`
 - `long double`
 - `char*` (`char_ptr`)
+
+### CTEST_DEFINE_ENUM_TYPE
+
+`CTEST_DEFINE_ENUM_TYPE` can be used with and enum defined by using the macro utils feature `MU_DEFINE_ENUM` in order to generate the needed `_ToString` and `_Compare` functions.
```
Comment (mattdurak) [inline doc/ctest_user_manual.md]: >and [](start = 42, length = 3)

"an", here and below

File: inc/ctest.h [hunk 2]
```diff
@@ -331,6 +331,30 @@ do \
 } \
 while(0)
 
+// this macro expans to the needed _ToString and _Compare functions for an enum,
```
Comment (mattdurak) [inline inc/ctest.h]: >expans [](start = 14, length = 6)

"expands" (here and below)

File: doc/ctest_user_manual.md (hunk 1 above)
Comment (dcristoloveanu) [reply doc/ctest_user_manual.md]: Done.

---
In reply to: [326827510](https://github.com/Azure/azure-ctest/pull/44#discussion_r326827510) [](ancestors = 326827510)

File: inc/ctest.h (hunk 2 above)
Comment (dcristoloveanu) [reply inc/ctest.h]: Done.

---
In reply to: [326827609](https://github.com/Azure/azure-ctest/pull/44#discussion_r326827609) [](ancestors = 326827609)

**Pull request merged**